src/together/filemanager.py

```diff
@@ -316,6 +327,16 @@ def upload(
         purpose: FilePurpose,
         redirect: bool = False,
     ) -> FileResponse:
+        if file.suffix == ".jsonl":
+            filetype = FileType.jsonl
+        elif file.suffix == ".parquet":
+            filetype = FileType.parquet
+        else:
+            raise FileTypeError(
+                f"Unknown extension of file {file}. "
+                "Only files with extensions .jsonl and .parquet are supported."
+            )
+
```

Comment:
Sure, one question though: do we want to avoid this check even if the user supplies a file of the type that's not supported? In other words, are we OK with the user spending time on uploading the file just to learn that it's incorrect?

Also, currently `redirect` seems to be always True when running the `upload command`: is this by design?

Reply:
> Sure, one question though: do we want to avoid this check even if the user supplies a file of the type that's not supported? In other words, are we OK with the user spending time on uploading the file just to learn that it's incorrect?

hm good point. but a different remote server might support more filetypes than just jsonl/parquet (like our dev server or another files API server). We don't want to break existing OAI compat.

> Also, currently redirect seems to be always True when running the upload command: is this by design?

This is a limitation on the server side. I'm working on fixing this so we don't need the redirect stuff anymore, hopefully later this week.